plugins/user-settings/src/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ChangeEvent, RefObject, useEffect, useState } from 'react';
+import {
+  Card,
+  CardContent,
+  CardHeader,
+  makeStyles,
+  PopoverActions,
+} from '@material-ui/core';
+import { AppSettingsList } from './AppSettingsList';
+import { AuthProvidersList } from './AuthProviderList';
+import { FeatureFlagsList } from './FeatureFlagsList';
+import { SignInAvatar } from './SignInAvatar';
+import { UserSettingsMenu } from './UserSettingsMenu';
+import { useUserProfile } from './useUserProfileInfo';
+import {
+  useApi,
+  featureFlagsApiRef,
+  TabbedCard,
+  CardTab,
+  InfoCard,
+} from '@backstage/core';
+import { DefaultProviderSettings } from './DefaultProviderSettings';
+import { ProviderSettings } from './UserSettings';
+import { Alert } from '@material-ui/lab';
+
+const useStyles = makeStyles({
+  root: {
+    minWidth: 400,
+    maxWidth: 500,
+  },
+});
+
+type Props = {
+  popoverActionRef: RefObject<PopoverActions | null>;
```

Comment:
Is it kosher to send a ref from one thing onto the other? 😬

I'd rather expect the caller to useCallback memoize the function and pass it directly, and for this component to store it in a ref of its own.